### PR TITLE
Grant write permissions only when a maintainer triggered a CI

### PR DESCRIPTION
**Pull request by hibariya**
See #1613 for the details.

### CODE REVIEW

**Review by ANGELAFN75 — CHANGES_REQUESTED** [review]
Good

**Review by cjavilla-stripe — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -2,30 +2,52 @@ name: CI
...
-    # branches:
-    #   - 'dependabot/**'
+      - 'main'
+  pull_request_target:
 
 env:
   STRIPE_PUBLISHABLE_KEY: ${{ secrets.TEST_STRIPE_PUBLISHABLE_KEY }}
   STRIPE_SECRET_KEY: ${{ secrets.TEST_STRIPE_SECRET_KEY }}
   COMPOSE_FILE: docker-compose.yml:docker-compose.playwright.yml
+  MAINTAINERS: |
```
Comment (paulasjes-stripe) [inline .github/workflows/ci.yml:12]: Can we automate getting this list?

**Review by paulasjes-stripe — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hideokamoto-stripe) [reply .github/workflows/ci.yml:12]: I create the issue to take over it.
https://github.com/stripe-samples/accept-a-payment/issues/1773

**Pull request merged**